src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,202 @@
+package duke.parser;
+
+import duke.command.Command;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.EventCommand;
+import duke.command.DeadlineCommand;
+import duke.command.TodoCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ListCommand;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
+    private static final String UNKNOWN_ERROR = "OOPS!!! I'm sorry, but I don't know what that means :-(";
+    private static final String EXTRA_ERROR = "The command contains extra information!";
+    private static final String INCOMPLETE_ERROR = "The command is too short and incomplete!";
+    private static final String NUMBER_ERROR = "Please give the number of task!";
+    private static final String DATETIME_ERROR = "Please follow this format \"31/01/2020 2300\" for date and time!";
+    private static final DateTimeFormatter IN_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Parse user input into command for execution.
+     *
+     * @param fullCommand full user input string
+     * @return the command based on the user input
+     */
+    public static Command parse(String fullCommand) {
+        String[] command = fullCommand.split(" ", 2);
+        switch (command[0]) {
+
+        case "list":
+            return prepareList(command);
+
+        case "done":
+            return prepareDone(command);
+
+        case "delete":
+            return prepareDelete(command);
+
+        case "todo":
+            return prepareTodo(command);
+
+        case "deadline":
+            return prepareDeadline(command);
+
+        case "event":
+            return prepareEvent(command);
+
+        case "find":
+            return prepareFind(command);
+
+        case "bye":
+            return prepareExit(command);
+
+        default:
+            return new InvalidCommand(UNKNOWN_ERROR);
+        }
```

Comment:
i suggest that you move the logic of these individual functions into the respective command classes' constructors. your Parser class is way too long right now, making it hard to extend if you want to add more commands.

Reply:
Thanks for your suggestion, I will consider changing my structure of parser and command class.